public/app/plugins/datasource/cloudwatch/components/ConfigEditor.test.tsx

```diff
@@ -104,53 +95,138 @@ describe('Render', () => {
   });
 
-  it('should show access key and secret access key fields', () => {
+  it('should show access key and secret access key fields when the datasource has not been configured before', async () => {
     setup({
       jsonData: {
         authType: AwsAuthType.Keys,
       },
     });
-    expect(screen.getByLabelText('Access Key ID')).toBeInTheDocument();
-    expect(screen.getByLabelText('Secret Access Key')).toBeInTheDocument();
+    await waitFor(async () => {
+      expect(screen.getByLabelText('Access Key ID')).toBeInTheDocument();
+      expect(screen.getByLabelText('Secret Access Key')).toBeInTheDocument();
+    });
   });
 
-  it('should show arn role field', () => {
+  it('should show arn role field', async () => {
     setup({
       jsonData: {
         authType: AwsAuthType.ARN,
       },
     });
-    expect(screen.getByLabelText('Assume Role ARN')).toBeInTheDocument();
+    await waitFor(async () => expect(screen.getByLabelText('Assume Role ARN')).toBeInTheDocument());
   });
 
-  it('should load log groups when multiselect is opened', async () => {
+  it('should display log group selector field', async () => {
     setup();
-    const multiselect = await screen.findByLabelText('Log Groups');
-    selectEvent.openMenu(multiselect);
-    expect(await screen.findByText('logGroup-foo')).toBeInTheDocument();
+    await waitFor(async () => expect(await screen.getByText('Select Log Groups')).toBeInTheDocument());
+  });
+
+  it('should only display the first two default log groups and show all of them when clicking "Show all" button', async () => {
+    setup({
+      version: 2,
+      jsonData: {
+        logGroups: [
+          { arn: 'arn:aws:logs:us-east-2:123456789012:log-group:logGroup-foo:*', name: 'logGroup-foo' },
+          { arn: 'arn:aws:logs:us-east-2:123456789012:log-group:logGroup-bar:*', name: 'logGroup-bar' },
+          { arn: 'arn:aws:logs:us-east-2:123456789012:log-group:logGroup-baz:*', name: 'logGroup-baz' },
+        ],
+      },
+    });
+    await waitFor(async () => {
+      expect(await screen.getByText('logGroup-foo')).toBeInTheDocument();
+      expect(await screen.getByText('logGroup-bar')).toBeInTheDocument();
+      expect(await screen.queryByText('logGroup-baz')).not.toBeInTheDocument();
+
+      await userEvent.click(screen.getByText('Show all'));
+
+      expect(await screen.getByText('logGroup-baz')).toBeInTheDocument();
+    });
+  });
+
+  it('should should save the data source if it wasnt saved before', async () => {
+    const ID = 1000;
+    const UNSAVED_VERSION = 1;
+    const SAVED_VERSION = 2;
+    const store = configureStore();
+    const newProps = {
+      ...props,
+      options: {
+        ...props.options,
+        id: ID,
+        version: UNSAVED_VERSION,
+      },
+    };
+
+    putMock.mockImplementation(async () => {
+      return {
+        datasource: {
+          version: SAVED_VERSION,
+        },
+      };
+    });
+
+    render(
+      <Provider store={store}>
```

Comment:
are we using this provider/store? Sorry I think I missed where that happens!

Reply:
No that's true, not necessary for this scenario. Removed it